src/basic/chattr-util.c

```diff
@@ -154,7 +154,7 @@ int read_attr_path(const char *p, unsigned *ret) {
         assert(p);
         assert(ret);
 
-        fd = open(p, O_RDONLY|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW);
+        fd = open(p, O_PATH|O_CLOEXEC|O_NOFOLLOW);
```

Comment:
Are you sure that ioctl() still works with an O_PATH fd?

Reply:
mm no, I tested only fstat()
```
openat(AT_FDCWD, "open", O_RDONLY|O_PATH) = 3
fstat(3, {st_mode=S_IFREG|0755, st_size=16768, ...}) = 0
ioctl(3, FS_IOC_GETFLAGS, 0x7ffd48b518fc) = -1 EBADF (Bad file descriptor)
```

I'll drop this chunk.
I testing the occurrences one by one, I'll leave this PR as draft until ready.